context-propagation-java5/src/main/java/nl/talsmasoftware/context/executors/ContextAwareExecutorService.java

```diff
@@ -39,7 +39,7 @@
  * @author Sjoerd Talsma
  */
 public class ContextAwareExecutorService extends CallMappingExecutorService {
-    private final Logger logger = Logger.getLogger(getClass().getName());
+    private static final Logger logger = Logger.getLogger(ContextAwareExecutorService.class.getName());
```

Comment:
I'd like this class to remain open for subclassing.
In that case, the logging _should_ probably go to the logger for the actual class (in my opinion).

Do you have a strong reason to use the superclass logger in that case?

Reply:
It appears there is in general some disagreement between whether loggers should be static or per instance. Though the most common reason for making them instance-based seems to be to support containers (e.g. J2EE), see [Apache Commons Logging / StaticLog](https://cwiki.apache.org/confluence/display/COMMONS/Logging+StaticLog) and [SLF4J FAQ](http://www.slf4j.org/faq.html#declared_static). I am not sure if these problems are still the case or if logging frameworks have implemented solutions for this already.

Making logger fields static has the advantage that it is likely more performance (CPU and memory-wise).

The disadvantage of using `getClass().getName()` is also that it might be confusing, consider the type hierarchy:
- `MyCustomExecutor`
  - `MyExecutor`
    - `ContextAwareExecutorService`
 
When an instance of `MyCustomExecutor` now runs a method from `ContextAwareExecutorService` which logs something, the log entry will claim that it came from the logger for `MyCustomExecutor`, which will be confusing.

Though it might have also some advantages.
However, it is not consistent across this project, `ServletRequestContextFilter` uses a static logger even though it can be subclassed.

Should I either consistently change it to use static or instance loggers when the declaring class can be subclassed?
(I would then adjust `AbstractThreadLocalContext` as well)

Reply:
I guess you're right about the inconsistency. I have no issues making the logger in `ServletRequestContextFilter` a non-static field as well. Although that class is not actually _designed for_ extension, it _is_ extendable in reality.

Reply:
The main point of my comment was rather why I prefer a static logger. There are probably now these options (1, 1.i, 2, 2.i, 3), which one do you prefer?
1. Keep the logger code as is (i.e. `ContextAwareExecutorService.logger` remains an instance field)
    1. Additionally make `ServletRequestContextFilter.logger` an instance field as well and use `getClass().getName()`
2. Keep `ContextAwareExecutorService.logger` as instance field but use `ContextAwareExecutorService.class` (addresses some of the concerns listed in my comment above)
    1. Additionally make `ServletRequestContextFilter.logger` an instance field but use `ServletRequestContextFilter.class`
3. Make `ContextAwareExecutorService.logger` static